### ISSUE TITLE
TM Request In Callback breaking async flow

### ISSUE THREAD

**Issue by ionutrazvanionita** [OpenSIPS/opensips#810]
TM calls Request In callback after the first async call, when the transaction is created, but there are some scenarios in which we need that callback to be called after all the async routes were called. Consider the following scenario:

```
route {
...
        if (is_method("INVITE")) {
                 /* transaction is created right after radius_send_auth
                  * and also REQUEST_IN callback is called*/
                 async(radius_send_auth(...), async_route);
        }
...
}

route[async_route] {
        /* user successfully authenticated */
        if ($rc==1) {
                  /* log cdrs for this user */
                  setflag(LOG_CDR);
        }
}

```

In this scenario we won't have any CDR's generated because when the request in callback shall be called, we won't have the LOG_CDR flag set.

Edit:

Reported by Dragomir Haralambiev

**Comment by ionutrazvanionita** [comment]
The problem was first reported in #799

**Comment by ionutrazvanionita** [comment]
fixed with pr #840 and #841
